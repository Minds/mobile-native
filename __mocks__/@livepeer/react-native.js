// mock @livepeer/react-native
export const LivepeerConfig = {
  setConfig: jest.fn(),
  getConfig: jest.fn(),
  get: jest.fn(),
  set: jest.fn(),
};

export const createReactClient = jest.fn();
export const createLivepeerClient = jest.fn();
export const Player = jest.fn();
