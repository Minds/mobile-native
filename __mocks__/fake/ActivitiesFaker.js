function load(count, container) {
  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid: code,
      custom_data: false,
      custom_type: false,
      rowKey: 'something' + i,
      description: 'Congratulations! ',
      edited: '',
      guid: code,
      mature: false,
      time_created: '1522036284',
      ownerObj: {
        guid: '824853017709780997',
        type: 'user',
        subtype: false,
        time_created: '1522036284',
      },
      message: 'Message',
      title: 'TITLE',
      owner_guid: '824853017709780997',
      parent_guid: '838106762591510528',
      perma_url: false,
      thumbnail_src: false,
      type: 'activity',
      wire_totals: {
        tokens: 1000000000000000000,
      },
      containerObj: container,
    };
  });

  return {
    activities: activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa',
  };
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load };
}

/**
 * Generate a single fake entity
 * @param {object} containerObj optional entity container
 * @returns
 */
export function fakeOne(containerObj) {
  const resp = load(1, containerObj);
  return resp.activities[0];
}
