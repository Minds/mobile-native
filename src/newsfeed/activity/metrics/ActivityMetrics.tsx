import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { observer } from 'mobx-react';
import Icon from 'react-native-vector-icons/Ionicons';
import McIcon from 'react-native-vector-icons/MaterialCommunityIcons';

import { CommonStyle } from '../../../styles/Common';
import abbrev from '../../../common/helpers/abbrev';
import token from '../../../common/helpers/token';
import number from '../../../common/helpers/number';
import ThemedStyles from '../../../styles/ThemedStyles';
import type ActivityModel from '../../../newsfeed/ActivityModel';

type PropsType = {
  entity: ActivityModel;
};

/**
 * Activity metrics component
 */
@observer
export default class ActivityMetrics extends Component<PropsType> {
  /**
   * Render
   */
  render() {
    const entity = this.props.entity;

    if (!entity.wire_totals) {
      return <View />;
    }

    return (
      <View style={[CommonStyle.rowJustifyCenter]}>
        <View
          style={[
            CommonStyle.rowJustifyStart,
            CommonStyle.borderRadius4x,
            CommonStyle.border,
            CommonStyle.borderHair,
            CommonStyle.borderGreyed,
            CommonStyle.paddingLeft,
            CommonStyle.paddingRight,
            ThemedStyles.style.backgroundTertiary,
            styles.container,
          ]}>
          <View style={[CommonStyle.rowJustifyCenter, CommonStyle.alignCenter]}>
            <Text style={styles.counter}>
              {abbrev(token(entity.wire_totals.tokens), 0)}{' '}
              <Icon name="ios-flash" />
            </Text>
          </View>
          <View style={[CommonStyle.rowJustifyCenter, CommonStyle.alignCenter]}>
            <Text style={[styles.counter]}> Â· </Text>
            <Text style={styles.counter}>
              {number(entity.impressions, 0)} <McIcon name="eye" />
            </Text>
          </View>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  counter: {
    alignItems: 'center',
    fontSize: 11,
  },
  container: {
    paddingVertical: 2,
    borderBottomWidth: 0,
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: 0,
  },
});
