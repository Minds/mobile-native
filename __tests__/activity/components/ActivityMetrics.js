import 'react-native';
import React from 'react';

import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';

import renderer from 'react-test-renderer';
import ActivityMetrics from '../../../src/newsfeed/activity/metrics/ActivityMetrics';

jest.mock('../../../src/auth/UserStore');

describe('activity metrics component', () => {
  let screen;
  beforeEach(() => {
    let activityResponse = activitiesServiceFaker().load(1);
    let activity = activityResponse.activities[0];
    activity.wire_totals = { tokens: 20 };
    activity.impressions = 20;
    screen = renderer.create(<ActivityMetrics entity={activity} />);
  });

  it('renders correctly', async () => {
    expect(screen).toMatchSnapshot();
  });

  it('should have Text', async () => {
    let root = screen.root;
    expect(root.findAllByType('Text')).toHaveLength(1);
  });
});
