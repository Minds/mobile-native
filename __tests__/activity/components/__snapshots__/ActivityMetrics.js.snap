// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity metrics component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-start",
        },
        Object {
          "borderRadius": 8,
        },
        Object {
          "borderWidth": 1,
        },
        Object {
          "borderWidth": 0.5,
        },
        Object {
          "borderColor": "#cccccc",
        },
        Object {
          "paddingLeft": 5,
        },
        Object {
          "paddingRight": 5,
        },
        undefined,
        Object {
          "borderBottomLeftRadius": 0,
          "borderBottomRightRadius": 0,
          "borderBottomWidth": 0,
          "paddingVertical": 2,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "fontSize": 11,
          }
        }
      >
        2e-17
         
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignItems": "center",
              "fontSize": 11,
            },
          ]
        }
      >
         · 
      </Text>
      <Text
        style={
          Object {
            "alignItems": "center",
            "fontSize": 11,
          }
        }
      >
        20
         
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          󰈈
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
